### PR TITLE
feat: add remotion autocompletion to fig

### PR DESCRIPTION
**Pull request by MatteoGauthier**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**

This PR add [remotion](https://remotion.dev) autocompletion

- [x] I’ve reduced the length of arguments label from "entry-file" to "entry" in order to fit in fig popup
- [x] `npx remotion render`
- [x] `npx remotion still`
- [x] `npx remotion preview`

### CODE REVIEW

File: src/npx.ts [hunk 1]
```diff
@@ -57,6 +57,11 @@ const suggestions: Fig.Suggestion[] = [
     name: "create-next-app",
     icon: "https://nextjs.org/static/favicon/favicon-16x16.png",
   },
+  {
+    name: "remotion",
+    icon:
+      "https://raw.githubusercontent.com/remotion-dev/remotion/main/packages/docs/static/img/logo-small.png",
+  },
```
Comment (fedeci) [inline src/npx.ts:64]: Should we add it to `yarn` spec too?

Comment (MatteoGauthier) [reply src/npx.ts:64]: yes maybe, what's the command i tried `yarn exec` and `yarn x` can't find a similar way to execute binary in yarn (like npx)

Comment (fedeci) [reply src/npx.ts:64]: Just add it here https://github.com/withfig/autocomplete/blob/5d285e0e9e971a36dc6ba9c1c40162c672e973b2/src/yarn.ts#L9

Comment (MatteoGauthier) [reply src/npx.ts:64]: Okay thanks, but how to use it in the terminal ?

Comment (fedeci) [reply src/npx.ts:64]: You need to be in a repo that has the cli tool installed and then run "yarn cmd"

**Comment by QuiiBz** [comment]
Hi, thanks for contributing! I converted this PR to draft since `npx remotion lambda` is waiting to be done, feel free to mark it "Ready for review" when it's done!

**Comment by MatteoGauthier** [comment]
> Hi, thanks for contributing! I converted this PR to draft since `npx remotion lambda` is waiting to be done, feel free to mark it "Ready for review" when it's done!

Okay thanks you !

File: src/npx.ts (hunk 1 above)
Comment (MatteoGauthier) [reply src/npx.ts:64]: Ok I will add that thanks

**Comment by MatteoGauthier** [comment]
I've removed `npx remotion lambda` to the checklist `remotion` creator didn't want this feature implemented for now.

File: src/remotion.ts [hunk 2]
```diff
@@ -0,0 +1,304 @@
...
+  subcommands: [
+    {
+      name: "render",
+      priority: 60,
+      description:
+        "Render a video based on the entry point, the composition ID and save it to the output location",
+      args: [
+        {
+          name: "entry",
+          isOptional: false,
```
Comment (QuiiBz) [inline src/remotion.ts:13]: You can remove `isOptional` when the value is `false` - it’s the default one

File: src/remotion.ts [hunk 3]
```diff
@@ -0,0 +1,304 @@
...
+        },
+        {
+          name: "--concurrency",
+          description: "How many frames to render in parallel",
+        },
+        {
+          name: "--image-format",
+          description: 'Format to render the frames in, "jpeg" or "png"',
+          args: {
+            template: ["filepaths"],
```
Comment (QuiiBz) [inline src/remotion.ts:56]: Shouldn’t this argument have `suggestions` instead of this `template`?

File: src/remotion.ts [hunk 4]
```diff
@@ -0,0 +1,304 @@
...
+          description: "Custom location for a Remotion config file",
+          args: {
+            template: "filepaths",
+          },
+        },
+        {
+          name: "--quality",
+          description: "Quality for rendered frames, JPEG only, 0-100",
+          args: {
+            default: "80",
```
Comment (QuiiBz) [inline src/remotion.ts:85]: I think it would be useful to have suggestions for 0, 10, 20… 80, 90, 100

File: src/remotion.ts [hunk 5]
```diff
@@ -0,0 +1,304 @@
...
+        {
+          name: "--browser-executable",
+          description: "Custom path for browser executable",
+          args: {
+            template: "filepaths",
+          },
+        },
+        {
+          name: "--frames",
+          description: "Render a portion or a still of a video, 0-9, 50",
```
Comment (QuiiBz) [inline src/remotion.ts:128]: Does this option have an argument?

File: src/remotion.ts [hunk 6]
```diff
@@ -0,0 +1,304 @@
...
+              { name: "error" },
+              { name: "warning" },
+              { name: "verbose" },
+              { name: "info" },
+            ],
+          },
+        },
+        {
+          name: "--port",
+          description: "Custom port to use for the HTTP server",
```
Comment (QuiiBz) [inline src/remotion.ts:150]: Same as above

File: src/remotion.ts [hunk 7]
```diff
@@ -0,0 +1,304 @@
...
+        },
+        {
+          name: "output",
+          isOptional: false,
+          template: ["filepaths"],
+          suggestions: ["out.mp4"],
+        },
+      ],
+      options: [
+        { name: "--frame", description: "Which frame to render (default 0)" },
```
Comment (QuiiBz) [inline src/remotion.ts:191]: Same as above

File: src/remotion.ts [hunk 8]
```diff
@@ -0,0 +1,304 @@
...
+        {
+          name: "--image-format",
+          description: 'Format to render the frames in, "jpeg" or "png"',
+          args: {
+            template: ["filepaths"],
```
Comment (QuiiBz) [inline src/remotion.ts:196]: Wrong argument?

**Review by QuiiBz — CHANGES_REQUESTED** [review]
Love remotion, thanks for submitting a spec for it!

**Comment by MatteoGauthier** [comment]
> Love remotion, thanks for submitting a spec for it!

Thanks for your proposals, loved to work on this !

**Review by QuiiBz — APPROVED** [review]
LGTM, thanks a lot for this contribution 🎉

**Comment by JonnyBurger** [comment]
Thank you @MatteoGauthier and @QuiiBz! 🥳 Already in the Terminal, so cool!

**Pull request merged**